tools/rosbag/src/record.cpp

```diff
@@ -62,6 +62,7 @@ rosbag::RecorderOptions parseOptions(int argc, char** argv) {
       ("bz2,j", "use BZ2 compression")
       ("lz4", "use LZ4 compression")
       ("split", po::value<int>()->implicit_value(0), "Split the bag file and continue recording when maximum size or maximum duration reached.")
+      ("max-splits", po::value<int>()->default_value(0), "Split the bag file at most splits times, then begin erasing the oldest split to keep a constant number of files.")
```

Comment:
Since the new options isn't doing the actual splitting but limits the number of log files I would suggest a help text like this:

> Keep a maximum of N bag files, when reaching the maximum erase the oldest one to keep a constant number of files.

The same / similar for the Python help string.